joomla/joomla-cms
# [4.3] Template Diff View

Pull Request for Issue multiple .

### Summary of Changes
Replace the very limited javascript used to display the diff between the core and the overriden file in com_templates with something much more useful


### Testing Instructions
Apply this PR and then composer install and npm i
or use on of the prebuilt packages

There are two different views for the override: side by side and inline
side by side is the default - you can swtich to the inline version in the component options
both show the line numbers


A message is displayed if the override is identical to the core original file


### Actual result BEFORE applying this Pull Request
too ugly to display ;)


## Expected result AFTER applying this Pull Request
### Default Side By Side View
![image](https://user-images.githubusercontent.com/1296369/191976508-3d034aab-323e-441d-acf2-e4b8351e85eb.png)

### Optional Inline View
![image](https://user-images.githubusercontent.com/1296369/191976163-a8107695-d0bf-427a-a94c-57e88f113725.png)

### Identical file View
![image](https://user-images.githubusercontent.com/1296369/191993761-7571a4cc-bfc5-40e9-aa8b-74808b4b97f3.png)

### Visible display of tabs and spaces
![image](https://user-images.githubusercontent.com/1296369/192152317-4722e35f-fa99-42c3-b1b3-3dffc7fa7ac0.png)

### Documentation Changes Required
probably just updated screenshot

### More Info
For more information on the library used and possible extras that could be added see https://github.com/jfcherng/php-diff

@richard67 a file is removed - shall I add it to the script or shall I ?

~There may be some tweaks to the build scripts required to remove stuff from the library but I can only check when the package is built here ;)~
